### PR TITLE
feat(workspace/user): add new resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support for creating workspace users through terraform.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new user resource support.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/workspace' TESTARGS='-run=TestAccUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run=TestAccUser_basic -timeout 360m -parallel 4
=== RUN   TestAccUser_basic
=== PAUSE TestAccUser_basic
=== CONT  TestAccUser_basic
--- PASS: TestAccUser_basic (24.72s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 24.781s
```

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -289,8 +289,12 @@ github.com/chnsz/golangsdk/openstack/waf_hw/v1/premium_instances
 github.com/chnsz/golangsdk/openstack/waf_hw/v1/valuelists
 github.com/chnsz/golangsdk/openstack/waf_hw/v1/webtamperprotection_rules
 github.com/chnsz/golangsdk/openstack/waf_hw/v1/whiteblackip_rules
+<<<<<<< 0360b9936078b713cb83b266a500bac251f9d548
```
Comment (hmbig2) [inline vendor/modules.txt]: pls remove the conflict of merge

Comment (Lance52259) [reply vendor/modules.txt]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/workspace' TESTARGS='-run=TestAccUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run=TestAccUser_basic -timeout 360m -parallel 4
=== RUN   TestAccUser_basic
=== PAUSE TestAccUser_basic
=== CONT  TestAccUser_basic
--- PASS: TestAccUser_basic (568.08s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 568.158s
```

File: docs/resources/workspace_user.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  Changing this will create a new resource.
+
+* `email` - (Required, String) Specifies the email address of user. The value can contain `1` to `64` characters.
+
+* `description` - (Optional, String) Specifies the description of user. The maximum length is `255` characters.
+
+* `account_expires` - (Optional, String) Specifies the user's valid period configuration.
+  + Never expires: **0**.
+  + Expires at a certain time: account expires must in Milli-RFC3339 format like `yyyy-MM-ddTHH:mm:ss.000Z`.
```
Comment (ShiChangkuo) [inline docs/resources/workspace_user.md]: suggest to update the timestamp without millisecond

Comment (Lance52259) [reply docs/resources/workspace_user.md]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/workspace' TESTARGS='-run=TestAccUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run=TestAccUser_basic -timeout 360m -parallel 4
=== RUN   TestAccUser_basic
=== PAUSE TestAccUser_basic
=== CONT  TestAccUser_basic
--- PASS: TestAccUser_basic (559.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 559.440s
```

**Pull request merged**